1558 review page and tabs

guatemartin opened the pull request:
Have tried to use Semantic as much as possible to do the heavy lifting.  Hopefully I haven't disturbed the desktop portion too much by using Grid columns in some places and leveraging the Stackable property of a grid.

1. Removed ProgressBar for mobile.  Fits well in simple 2 step processes, but in the more complicated it goes awry.
2. Removed Icon in the Review->Overview Tab.   Didn't seem necessary to provide in mobile
3. Tab headers wrap.  Thought about drop down as it might be cleaner looking, but tabs are easier to navigate.
4. Checked assignment tab looks ok with multiple stages.

guatemartin commented on semantic/src/site/globals/site.overrides:
```diff
@@ -906,8 +906,7 @@ a:hover {
-    border: 1px solid @surfacesMed;
-    min-width: 400px;
+    border: 1px solid @surfacesMed;    
     h4 {
```
Removed this min-width site wide.  Couldn't see it being of any benefit anywhere, and it would stuff up the mobile rendering.

guatemartin commented on semantic/src/site/globals/site.overrides:
```diff
@@ -930,6 +929,7 @@ a:hover {
       margin-bottom: 5px;
       .ui.label {
         line-height: 1.5;
+        font-size: 0.57;
       }
       .centered-flex-box-row {
         display: flex;
```
Why 0.57?  That's just what seemed to look best.

CarlosNZ replied:
![Screenshot 2023-11-07 at 12 05 40 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/e4c3370d-c265-4234-ad59-749989c64e7f)

I reckon we can lose the "STAGE" label, cos it's clear from the coloured bubble that it's referring to stage, which will give more room for the other stuff.

And maybe, let it wrap so the "Stage" bubble can go to next line? (It'd be good if the text/arrow inside the drop-down can stay on a single line)

guatemartin replied:
Text Label gone.  Left as much spacing btn dropdown and Stage Label as I could.  It's not much really.
![assignment_tab](https://github.com/msupply-foundation/conforma-web-app/assets/17306152/f1dde0ff-3229-4e7d-9dcc-5709e72b3d1a)

guatemartin commented on src/containers/Review/assignment/ReviewSectionRow.tsx:
```diff
@@ -87,13 +87,19 @@ const ReviewSectionRow: React.FC<ReviewSectionRowProps> = ({
       {canRenderRow && (
-        <Grid className="flex-row" verticalAlign="middle">
-          <Grid.Row columns={4} className="reviewer-row">
+        <Grid stackable columns={4} verticalAlign="middle">
+          <Grid.Column>
             <ReviewSectionRowAssigned {...props} />
+          </Grid.Column>
+          <Grid.Column>
             <ReviewSectionRowLastActionDate {...props} />
+          </Grid.Column>
+          <Grid.Column>
             <ReviewSectionRowProgress {...props} />
+          </Grid.Column>
+          <Grid.Column>
             <ReviewSectionRowAction {...props} />
-          </Grid.Row>
+          </Grid.Column>
         </Grid>
```
I thought I would try to use Semantic's Grid Stackable property if possible.  If it's not what we want, I can find another way.

CarlosNZ replied:
![Screenshot 2023-11-07 at 12 12 29 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/9d6f9489-be89-42f0-a442-a1fdb5f1eebe)
Yeah that's great, thanks. Only thing I'd suggest is for the rows within each block to be a bit closer together.

guatemartin replied:
Reduced spacing... we'll see if it's enough, or needs more...

CarlosNZ commented on semantic/src/site/collections/menu.overrides:
```diff
@@ -1,3 +1,10 @@
\ No newline at end of file
+*******************************/
+
+@media only screen and (max-width: @largestMobileScreen) {
+  // For review page tabs when mobile, but probably good site wide
+  .ui.attached.tabular.menu {
+    flex-wrap: wrap;
+  }
+}
```
Good to be able to wrap the tabs. However, the upper ones look a little weird just "floating" above the rest of the row and not attached to the container beneath.

Not sure what the solution is. It's a slight improvement if you add the border to the bottom (it only goes on left, right, top normally), but still doesn't look great. Would it be reasonable to *switch* the order so that the "active" one was always on the bottom row, or would that be annoying?

Any other suggestions?

CarlosNZ commented on src/containers/Review/overview/Overview.tsx:
```diff
@@ -29,12 +30,13 @@ export const Overview: React.FC<{
-          <Icon name="info circle" color="teal" />
+          {!isMobile && <Icon name="info circle" color="teal" />}
           <Message.Content>
             <div className="flex-row-center wrap">
```
![Screenshot 2023-11-07 at 12 16 18 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/d66bee03-77f5-4192-be72-ccebeb4b9e2d)

Just need to get the "Extend deadline" UI to wrap on mobile. (Hit me up in Telegram if you can't find an application with this set -- or just change the condition to `true` temporarily to make it show up.)

guatemartin replied:
Ohh, forgot this one...  Will get it

CarlosNZ reviewed (changes requested):
Cool, looking much better. Good call to use the Grid properly to stack nicely.

See comments for some small adjustments.

Also, the Notes and Documents tabs could do with a little tidy up -- you want to make a new issue for them?
 
![Screenshot 2023-11-07 at 12 21 12 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/1bdb75b2-d47e-46d3-b2fa-8ec0b5f77e4f)
- Left justify sort/files UI
- Input box full width

![Screenshot 2023-11-07 at 12 21 54 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/c7f2871d-6215-4ec4-8f8d-dff9812a8b81)- 
- icon a bit smaller
- reduce spacing

CarlosNZ commented on src/containers/Review/ReviewWrapper.tsx:
```diff
@@ -105,7 +107,7 @@ const ReviewWrapper: React.FC<ReviewWrapperProps> = ({ structure }) => {
           />
           <div id="review-home-content">
-            <ReviewProgress structure={structure} />
+            {!isMobile && <ReviewProgress structure={structure} />}
```
Oh one more thing. The progress bar actually looks okay on mobile up to 3 items. Maybe we should keep it unless there are more that two stages (which makes 3 dots in the bar)?

![Screenshot 2023-11-07 at 12 28 33 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/2c787ae9-4a6e-4449-9c89-b39602eab1f3)

(But just reduce the top/bottom space around it)

CarlosNZ replied:
Actually, don't worry about it. It's still a bit cut off. More trouble than it's worth I think.

guatemartin replied:
Oh, I actually did this.  If more than 2 stages, will not show in mobile.

CarlosNZ commented:
Oh, couple of other things I just noticed after posting my review:

There is a min-width of 300px on the `#overview .left-item` class that is causing problems when we get narrow:
![Screenshot 2023-11-07 at 12 37 09 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/9e18e5a0-34ca-420f-a22f-082d398b0bd0)
It's fine if you turn this off for mobile:

![Screenshot 2023-11-07 at 12 37 15 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/ec4606b6-ca51-4ce4-a33d-ff6f062b3c4b)


And can we reduce the padding around the Activity region to use more of the screen width (maybe drop the indents down a little too, if it helps?)

![Screenshot 2023-11-07 at 12 37 28 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/4254d03c-74d0-49f9-8abb-9db358b3c815)

I think we can also revert the font-size (0.57 rem) on the Stage bubbles back to what it was, as with the other tweaks I suggested it's probably not necessary.

CarlosNZ commented on semantic/src/site/globals/site.overrides:
```diff
@@ -2780,16 +2779,47 @@ h6:first-child {
 
   #notes-tab .options-row {
     flex-direction: column;
+    align-items: flex-start;
```
Can you just add some left padding to make this row flush with the stuff beneath it?
![Screenshot 2023-11-07 at 4 59 25 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/f98dec02-2b01-4093-91da-48c01f046984)

guatemartin commented on semantic/src/site/collections/message.overrides:
```diff
@@ -38,3 +38,9 @@
+
+@media only screen and (max-width: @largestMobileScreen) {
+  .ui.message {
+    padding: 0.5em 0.5em;
```
With the removal of other properties that were limiting the space... this is now 1em, and it isn't causing issues.

CarlosNZ commented on semantic/src/site/globals/site.overrides:
```diff
@@ -2780,16 +2779,47 @@ h6:first-child {
+    .stage-label {
+      margin-left: 3px;
+    }
+    .custom-assignment-grid-column {
```
![Screenshot 2023-11-07 at 5 11 03 PM](https://github.com/msupply-foundation/conforma-web-app/assets/5456533/13e640f5-c8a1-4826-9151-2b3c01c2ee98)

I think we need a bit more tidying up of this layout. But lets leave that to another issue.

guatemartin commented:
> Oh, couple of other things I just noticed after posting my review:
> 

I hadn't seen this one!  Thanks.  It was causing me grief and I couldn't find it!!

> There is a min-width of 300px on the `#overview .left-item` class that is causing problems when we get narrow: ![Screenshot 2023-11-07 at 12 37 09 PM](https://user-images.githubusercontent.com/5456533/280874888-9e18e5a0-34ca-420f-a22f-082d398b0bd0.png) It's fine if you turn this off for mobile:
> 
> ![Screenshot 2023-11-07 at 12 37 15 PM](https://user-images.githubusercontent.com/5456533/280874924-ec4606b6-ca51-4ce4-a33d-ff6f062b3c4b.png)
> 

It was actually the margins that were huge.  Have reduced for mobile.

> And can we reduce the padding around the Activity region to use more of the screen width (maybe drop the indents down a little too, if it helps?)
> 
> ![Screenshot 2023-11-07 at 12 37 28 PM](https://user-images.githubusercontent.com/5456533/280876322-4254d03c-74d0-49f9-8abb-9db358b3c815.png)
>

There was a little more space, but I still left it at 0.7rem, otherwise it uses up a lot of space.... I left it at .7 thinking it looked good, but if not, 1rem it is.
 
> I think we can also revert the font-size (0.57 rem) on the Stage bubbles back to what it was, as with the other tweaks I suggested it's probably not necessary.

CarlosNZ commented on src/containers/Review/ReviewWrapper.tsx:
```diff
@@ -107,7 +107,11 @@ const ReviewWrapper: React.FC<ReviewWrapperProps> = ({ structure }) => {
+              <ReviewProgress structure={structure} />
+            ) : (
+              !isMobile && <ReviewProgress structure={structure} />
+            )}
```
Sorry, I think I put you wrong. Let's just leave it how it was (hide on mobile) for now, and we can look at this a bit later. I think it needs a bit more before consideration it can work on mobile.

(I'm also thinking that it's generally going to be most useful when there are a lot of stages, so it's less useful for only 1-2 stage applications anyway)

CarlosNZ reviewed (approved):
Cool, working pretty well thanks. There's just a couple of tiny things if you could quickly fix before merging. And I'll make a new issue later on for a few additional tweaks once we've got used to this.

guatemartin commented:
@CarlosNZ lots of changes here

I just haven't done the "extend date" in overview, as I can't find one that has those properties.

In Overview 
- I removed the left coloured border as without the icon it makes no sense
- reduced the margin of the activities timeline
- removed '.left-area' class that had min-width 300px.  Not needed

Assignment Tab
- Removed the Text Label for the Stage
- Reduced the padding and spacing
- Increased Stage Label size
![assignment_tab](https://github.com/msupply-foundation/conforma-web-app/assets/17306152/1382a6f9-6580-4810-90ca-794de5ac61f5)

Notes Tab
- Left aligned dropdown and toggle button
- Textarea now 'stretches' to use up space
- Reduced spacing
![notes_tab_textarea](https://github.com/msupply-foundation/conforma-web-app/assets/17306152/60fa0238-6a85-417a-8202-1d892f832eb1)

Documents
- Smaller icon (60 x 60px) could go smaller
- Reduced padding and gap between files
![image](https://github.com/msupply-foundation/conforma-web-app/assets/17306152/05c372d1-82ee-4a37-980b-1d2dd45e6a3c)
